manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat' => [Package['snmp-client'], Package[$snmp::params::package_name]],
```

Comment:
@elisiano I considered that but I wanted to avoid hard coding the package name since it may change in the future and it's a lot easier update everything in params.pp (or hiera) when needed. IMO the first requirement should also be changed to Package[$snmp::params::client_package however I wanted to make as few changes as possible with this commit.